### PR TITLE
feat: Bitget wallet

### PR DESCRIPTION
**Pull request by abner-jie**
Hi, Team NEAR,

We have initiated PR again.

This PR adds bitget wallet to the list of supported wallets, to help our users connecting to wallet-selector dapps.

Please let me know if anything needs to be added or changed to satisfy your requirements.

Cheers,
Bitget wallet team

### CODE REVIEW

**Comment by kujtimprenkuSQA** [comment]
Hi, @abner-jie thanks for submitting this PR.

Can you please change the base branch from `main` to `dev`?
https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md#branching

File: packages/bitget-wallet/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const bitgetWallet = setupBitgetWallet({
+  iconUrl: bitgetWalletIconUrl
+});
+```
+
+## License
+
+This repository is distributed under the terms of both the MIT license and the Apache License (Version 2.0).
+ Binary file addedBIN +38.5 KB 
+packages/bitget-wallet/assets/bitget-wallet-icon.png
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/README.md:61]: Can you remove this text from here?

File: packages/bitget-wallet/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@near-wallet-selector/bitget-wallet",
+  "version": "8.5.3",
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/package.json:3]: Can you update the version here to match with the latest version of wallet-selector (v8.8.0)?

File: packages/bitget-wallet/src/lib/bitget-wallet.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    // browser extension background env.
+    // Check for isSignedIn() in only if extension is installed.
+    if (installed) {
+      await waitFor(() => !!window.bitkeep?.near?.isSignedIn()).catch(
+        () => false
+      );
+    }
+
+    return {
+      id: "bitgetWallet",
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/src/lib/bitget-wallet.ts:270]: Please change the wallet id to `bitget-wallet` the id must match the package-name.

File: packages/bitget-wallet/src/lib/bitget-wallet.ts [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+        publicKey: pubKey,
+        keyType: "0",
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.signMessage(
+        encoded,
+        accountId,
+        networkId
+      );
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/src/lib/bitget-wallet.ts:184]: This here is a bit confusing the public API of a wallet for `signMessage` should be implemented based on [NEP-413](https://github.com/near/NEPs/blob/master/neps/nep-0413.md).

If you have a look at the Sender Wallet there the signMessage of the signer is used to sing the payload for verifyOwner: https://github.com/near/wallet-selector/blob/main/packages/sender/src/lib/sender.ts#L242

The support for `signMessage` based on NEP-413 should be added in a similar way like this one https://github.com/near/wallet-selector/pull/997

File: packages/bitget-wallet/src/lib/bitget-wallet.ts [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    signOut,
+    async getAccounts() {
+      return getAccounts();
+    },
+    async verifyOwner({ message }) {
+      logger.log("BitgetWallet:verifyOwner", { message });
+
+      const networkId = options.network.networkId || "";
+      const accountId = (await _state.wallet.getAccountId()) || "";
+      const pubKey = (await _state.wallet.getPublicKey()) || "";
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/src/lib/bitget-wallet.ts:167]: Why is there a need to default to `""` for these variables, will the payload/message be signed correctly without any of them?

File: packages/bitget-wallet/src/lib/bitget-wallet.ts [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+      const data = {
+        accountId,
+        message,
+        blockId: block.header.hash,
+        publicKey: pubKey,
+        keyType: "0",
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.verifyOwner(
```
Comment (kujtimprenkuSQA) [inline packages/bitget-wallet/src/lib/bitget-wallet.ts:180]: I guess now the extension will need to be updated to add this function in `bitkeep.near.verifyOwner` right?

**Review by kujtimprenkuSQA — COMMENTED** [review]
This looks good to me, I left a question about verifyOwner in the comments below, and also wanted to ask if there's any way to enable `testnet` network for NEAR on Bitget Wallet?

File: packages/bitget-wallet/src/lib/bitget-wallet.ts (hunk 6 above)
Comment (abner-jie) [reply packages/bitget-wallet/src/lib/bitget-wallet.ts:180]: We will launch version 2.1.1 in the near future to support

**Comment by abner-jie** [comment]
Hi, @kujtimprenkuSQA 
thank you for your question, we have no plans to support `testnet` at the moment

**Review by kujtimprenkuSQA — APPROVED** [review]

File: packages/bitget-wallet/src/lib/bitget-wallet.ts (hunk 6 above)
Comment (kujtimprenkuSQA) [reply packages/bitget-wallet/src/lib/bitget-wallet.ts:180]: Looks like the new version was published on Chrome, I just tested and verifyOwner is working now too.

**Comment by kujtimprenkuSQA** [comment]
> Hi, @kujtimprenkuSQA thank you for your question, we have no plans to support `testnet` at the moment

Thank you for letting us know.

Is there any plan to add support for `signMessage` based on [NEP-413](https://github.com/near/NEPs/blob/master/neps/nep-0413.md)?

This function is very important in a new feature we plan to introduce in the wallet selector https://github.com/near/wallet-selector/pull/883

**Comment by kujtimprenkuSQA** [comment]
> > > Hi, @kujtimprenkuSQA thank you for your question, we have no plans to support `testnet` at the moment
> > 
> > 
> > Thank you for letting us know.
> > Is there any plan to add support for `signMessage` based on [NEP-413](https://github.com/near/NEPs/blob/master/neps/nep-0413.md)?
> > This function is very important in a new feature we plan to introduce in the wallet selector #883
> 
> Hi @kujtimprenkuSQA , we sill schedule support in future vesions. maybe next release.

That sounds great.

We published a new version and now Bitget Wallet is included in Wallet Selector:
https://www.npmjs.com/package/@near-wallet-selector/bitget-wallet

**Pull request merged**